spring-vault-core/src/main/java/org/springframework/vault/support/Certificate.java

```diff
@@ -39,103 +38,121 @@
-	 */
-	public static Certificate of(String serialNumber, String certificate, String issuingCaCertificate) {
-
-		Assert.hasText(serialNumber, "Serial number must not be empty");
-		Assert.hasText(certificate, "Certificate must not be empty");
-		Assert.hasText(issuingCaCertificate, "Issuing CA certificate must not be empty");
-
-		return new Certificate(serialNumber, certificate, issuingCaCertificate);
-	}
-
-	/**
-	 * @return the serial number.
-	 */
-	public String getSerialNumber() {
-		return this.serialNumber;
-	}
-
-	/**
-	 * @return encoded certificate (PEM or DER-encoded).
-	 */
-	public String getCertificate() {
-		return this.certificate;
-	}
-
-	/**
-	 * @return encoded certificate of the issuing CA (PEM or DER-encoded).
-	 */
-	public String getIssuingCaCertificate() {
-		return this.issuingCaCertificate;
-	}
-
-	/**
-	 * Retrieve the certificate as {@link X509Certificate}. Only supported if certificate
-	 * is DER-encoded.
-	 * @return the {@link X509Certificate}.
-	 */
-	public X509Certificate getX509Certificate() {
-
-		try {
-			byte[] bytes = Base64Utils.decodeFromString(getCertificate());
-			return KeystoreUtil.getCertificate(bytes);
-		}
-		catch (CertificateException e) {
-			throw new VaultException("Cannot create Certificate from certificate", e);
-		}
-	}
-
-	/**
-	 * Retrieve the issuing CA certificate as {@link X509Certificate}. Only supported if
-	 * certificate is DER-encoded.
-	 * @return the issuing CA {@link X509Certificate}.
-	 */
-	public X509Certificate getX509IssuerCertificate() {
-
-		try {
-			byte[] bytes = Base64Utils.decodeFromString(getIssuingCaCertificate());
-			return KeystoreUtil.getCertificate(bytes);
-		}
-		catch (CertificateException e) {
-			throw new VaultException("Cannot create Certificate from issuing CA certificate", e);
-		}
-	}
-
-	/**
-	 * Create a trust store as {@link KeyStore} from this {@link Certificate} containing
-	 * the certificate chain. Only supported if certificate is DER-encoded.
-	 * @return the {@link KeyStore} containing the private key and certificate chain.
-	 */
-	public KeyStore createTrustStore() {
-
-		try {
-			return KeystoreUtil.createKeyStore(getX509Certificate(), getX509IssuerCertificate());
-		}
-		catch (GeneralSecurityException | IOException e) {
-			throw new VaultException("Cannot create KeyStore", e);
-		}
-	}
+  private final String serialNumber;
+
+  private final String certificate;
+
+  private final String issuingCaCertificate;
+
+  private final CertificateEncoding certificateFormat;
+
+  private final CertificateEncoding issuingCaCertificateFormat;
+
+  Certificate(@JsonProperty("serial_number") String serialNumber, @JsonProperty("certificate") String certificate,
+      @JsonProperty("issuing_ca") String issuingCaCertificate) {
+
+    this.serialNumber = serialNumber;
+    this.certificate = certificate;
+    this.issuingCaCertificate = issuingCaCertificate;
+    this.certificateFormat = CertificateEncodingValidator.getFormat(certificate);
```

Comment:
We have no information about the current encoding so we have to calculate it for further processing. Should this an enum or a string only?